kafkian/consumer.py

```diff
@@ -54,29 +55,37 @@ def _subscribe(self):
         self._subscribed = True
 
     def __iter__(self):
+        self._subscribe()
         return self
 
     def __next__(self):
         self._subscribe()
         try:
             return next(self._generator)
-        except KafkaException:
-            raise StopIteration
```

Comment:
I don't know the `Consumer` api here very well, but I think if `Consumer` is an iterator, then at some point it should raise `StopIteration`. Is this correct?

Reply:
Not exactly. `StopIteration` is raised upon the "end of data" which it never the case with Kafka. We think of topics as endless streams, so the only reason for exiting the iterator is, in fact, the exception, and we prefer to explicitly raise it instead of silently ignoring it via raising StopIteration.